compiler/damlc/daml-stdlib-src/DA/Validation.daml

```diff
@@ -70,6 +70,11 @@ instance Applicative (Validation err) where
     Errors e1 *> Success _ = Errors e1
     Errors e1 *> Errors e2 = Errors (e1 <> e2)
 
+instance Semigroup (Validation err a) where
+    Errors e1 <> Errors e2 = Errors (e1 <> e2)
+    Success _ <> x = x
+    x <> Success _ = x
```

Comment:
this is not the same behavior as the Haskell version, which has

```haskell
instance Semigroup e => Semigroup (Validation e a) where
  x@Success{} <> _ = x -- important!
  _ <> x@Success{} = x -- important! 
  Failure e1 <> Failure e2 = Failure (e1 <> e2) 
```

the difference is that the Haskell version evaluates to the first `Success`, if any, while this one evaluates to the last `Success` only when there are no `Failure`s. 

I think the only justification for the Haskell version's behavior is that it allows `instance Monoid e => Monoid (Validation e a) where mempty = Failure mempty`, which isn't relevant here since the `Semigroup` is fixed to `NonEmpty`, which cannot be `Monoid` anyway. Thus I suggest to add the docstring from my other comment.

Reply:
Agreed. Thanks @akrmn!